apps/wallet/src/ui/app/pages/home/nft-transfer/index.tsx

```diff
@@ -1,124 +1,48 @@
-import { createAccountNftByIdSelector } from '_redux/slices/account';
-import { transferNFT } from '_redux/slices/sui-objects';
-import { DEFAULT_NFT_TRANSFER_GAS_FEE } from '_redux/slices/sui-objects/Coin';
-import PageTitle from '_src/ui/app/shared/PageTitle';
-
-import type { SerializedError } from '@reduxjs/toolkit';
-import type { FormikHelpers } from 'formik';
-
-import st from './TransferNFTForm.module.scss';
-
-const initialValues = {
-    to: '',
-};
-
-export type FormValues = typeof initialValues;
+import { NFTDisplayCard } from '_components/nft-display';
+import Overlay from '_components/overlay';
+import { useOwnedNFT } from '_hooks';
 
 function NftTransferPage() {
     const { nftId } = useParams();
-    const nftSelector = useMemo(
-        () => createAccountNftByIdSelector(nftId || ''),
-        [nftId]
-    );
-    const selectedNft = useAppSelector(nftSelector);
-    const objectId = selectedNft ? getObjectId(selectedNft) : null;
-    const address = useAppSelector(({ account: { address } }) => address);
-    const dispatch = useAppDispatch();
-    const [sendError, setSendError] = useState<string | null>(null);
-    const validationSchema = useMemo(
-        () => createValidationSchema(address || '', objectId || ''),
-        [address, objectId]
-    );
+    const address = useActiveAddress();
+
+    // verify that the nft is owned by the user and is transferable
+    const { data: ownedNFT, isLoading } = useOwnedNFT(nftId!, address);
     const navigate = useNavigate();
-    const onHandleSubmit = useCallback(
-        async (
-            { to }: FormValues,
-            { resetForm }: FormikHelpers<FormValues>
-        ) => {
-            if (!objectId) {
-                return;
-            }
-            setSendError(null);
-            try {
-                const resp = await dispatch(
-                    transferNFT({
-                        recipient: to,
-                        objectId,
-                        gasBudget: DEFAULT_NFT_TRANSFER_GAS_FEE,
-                    })
-                ).unwrap();
-                resetForm();
-                if (resp.txId) {
-                    navigate(
-                        `/receipt?${new URLSearchParams({
-                            txdigest: resp.txId,
-                            from: 'nfts',
-                        }).toString()}`
-                    );
-                }
-            } catch (e) {
-                setSendError((e as SerializedError).message || null);
-            }
-        },
-        [dispatch, navigate, objectId]
-    );
-    const handleOnClearSubmitError = useCallback(() => {
-        setSendError(null);
-    }, []);
-    const { loading } = useObjectsState();
-    return (
-        <div className={cl(st.container, { 'items-center': loading })}>
-            <Loading loading={loading}>
-                {selectedNft && objectId && hasPublicTransfer(selectedNft) ? (
```

Comment:
I think we should keep the `hasPublicTransfer` check just in case someone navigates to a token that is not transferable